Doc/library/imaplib.rst

```diff
@@ -307,6 +308,97 @@ An :class:`IMAP4` instance has the following methods:
    of the IMAP4 QUOTA extension defined in rfc2087.
 
 
+.. method:: IMAP4.idle(duration=None)
+
+   Return an :class:`!Idler`: an iterable context manager implementing the IMAP4 ``IDLE``
+   command as defined in :rfc:`2177`.
+
+   The returned object sends the ``IDLE`` command when activated by the
+   :keyword:`with` statement, produces IMAP untagged responses via the
+   :term:`iterator` protocol, and sends ``DONE`` upon context exit.
+
+   The *duration* argument sets a maximum duration (in seconds) to keep idling,
+   after which any ongoing iteration will stop.  It defaults to ``None``,
+   meaning no time limit.  Callers wishing to avoid inactivity timeouts on
+   servers that impose them should keep this at most 29 minutes.
+   See the :ref:`warning below <windows-pipe-timeout-warning>` if using
+   :class:`IMAP4_stream` on Windows.
+
+   Response tuples produced by the iterator almost exactly match those
+   returned by other methods in this module.  The difference is that the tuple's
+   second member is a single response datum, rather than a list of data.
+   Therefore, in a mailbox where calling ``M.response('EXISTS')`` would
+   return ``('EXISTS', [b'1'])``, the idle iterator would produce
+   ``('EXISTS', b'1')``.
+
+   Example::
+
+      with M.idle(duration=29 * 60) as idler:
+          for typ, datum in idler:
+              print(typ, datum)
+
+      ('EXISTS', b'1')
+      ('RECENT', b'1')
+
+   Instead of iterating one response at a time, it is also possible to retrieve
+   the next response along with any immediately available subsequent responses
+   (e.g. a rapid series of ``EXPUNGE`` events from a bulk delete).  This
+   batch processing aid is provided by the context object's ``burst()``
+   :term:`generator`:
+
+   .. method:: Idler.burst(interval=0.1)
+
+      Yield a burst of responses no more than *interval* seconds apart.
+
+      Example::
+
+         with M.idle() as idler:
```

Comment:
Ah sorry. Actually, it would be used as follows:

```rst

.. code-block:: pycon

   >>> print(2)
   2
```

This a syntax highlighting provided by Sphinx (based on reST). The docs are more than just reST documents, they are Sphinx-compatible. If you use the `.. code-block::` or `::` markers, you would have a default Python highlighting (as if you were writing a script) but with `pycon`, you would get the same highlighting as if you were writing stuff in the REPL.

I don't remember whether we automatically detect `>>>` for syntax highlighting or not, but it never hurts to be explicit.